src/python/example/alpha_complex_from_generated_points_example.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+
+from gudhi import random_point_generators
+from gudhi import AlphaComplex, SimplexTree
+from gudhi import plot_persistence_barcode, plot_persistence_diagram
+
+import matplotlib.pyplot as plt
+
+
+""" This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+    Author(s):       Hind Montassif
+
+    Copyright (C) 2021 Inria
+
+    Modification(s):
+      - YYYY/MM Author: Description of the modification
+"""
+
+__author__ = "Hind Montassif"
+__copyright__ = "Copyright (C) 2021 Inria"
+__license__ = "MIT"
+
+print("#####################################################################")
+print("AlphaComplex creation from generated points")
+
+
+# Generate a circle: 50 points; dim 2; radius 1
+points = random_point_generators.generate_points_on_sphere_d(50, 2, 1)
+
+# Plot the generated points (to uncomment if wished)
+#plt.scatter(points[:,0], points[:,1])
+#plt.show()
```

Comment:
You could also use a user argument flag to display or not points and diagram, like here:
https://github.com/GUDHI/gudhi-devel/blob/009c01c1d10c681fa441dbd4e4e166bbfd6f70e7/src/python/example/alpha_complex_diagram_persistence_from_off_file_example.py#L35-L40
...
https://github.com/GUDHI/gudhi-devel/blob/009c01c1d10c681fa441dbd4e4e166bbfd6f70e7/src/python/example/alpha_complex_diagram_persistence_from_off_file_example.py#L65-L68
No problem if the default behaviour is the contrary.

Reply:
I am always reluctant to add cruft to examples. I see examples as code meant to be read to see quickly how to use a function. Including them as inline snippets in the doc would be an option. If I look at alpha_complex_diagram_persistence_from_off_file_example.py, it has 73 lines, but less than 10 are relevant, all the argument parsing distracts the reader and makes it harder to find the important lines, and belongs more in a utility than an example.
Or maybe the short examples I am thinking about should be written inline in the doc (where they can also be tested), and examples/*.py is for bigger examples/utilities like alpha_complex_diagram_persistence_from_off_file_example.py? And we have tutorial notebooks for longer pieces of readable code? That would make sense and stop me from complaining about long examples :wink:
Note that it is possible to have a test that imports an example, and mocks the graphic functions, so that the example remains pretty and can still be run as a test. I think we do that in test_representations.py (although maybe not in the cleanest way). Not saying we have to do it, just a possibility.
Once we have a few generators, we will probably want a notebook showing them, with a plot of the point cloud and its diagram, unless we include it in existing notebooks.

Reply:
Ok, so maybe leave this example (to be run as test) as it is and remove the commented graphic part, that will be integrated later in the new notebook showing the different generators / or already existing ones.